### ISSUE TITLE
ALLOW BLANK SCREEN - option gnome 45

### ISSUE THREAD

**Issue by dom21121** [eonpatapon/gnome-shell-extension-caffeine#305]
Hi,

Great extension, but "ALLOW BLANK SCREEN" option doesn't work any more under gnome 45.

Thanks to consider.

:-)

**Comment by LogExE** [comment]
Yeah, computer suspends even if caffeine was enabled with this "allow blank screen" option. I'm using GNOME 45.1

**Comment by pakaoraki** [comment]
I confirm this problem on Gnome 45 / Ubuntu 23.10. Thanks for reporting.

**Comment by Fail-gh** [comment]
Same problem on Gnome 46 / Fedora 40

**Comment by pakaoraki** [comment]
Hi, It was recently working fine again on fedora 39 after testing few weeks ago. There is time to time issues with Fedora that seams to not manage inhibitors flag correctly for some reasons (similar to #311 ). Fedora 40 is still in beta, so I suggest to wait for the official release.

If the flag are correctly set by the extension, I'm not sure we can do something about this.

**Comment by Fail-gh** [comment]
On Fedora 39 I had the same problem, [Stimulator](https://flathub.org/apps/io.github.sigmasd.stimulator) worked and still works

**Comment by marshmn** [comment]
I'm having this issue on Fedora 41.

**Comment by colvdv** [comment]
I managed to manually fix the issue of Screen Blanking not working on Fedora 42. Here is how:

1. Open `~/.local/share/gnome-shell/extensions/caffeine@patapon.info/extension.js`.
2. Find this:
```
/* Inhibited flag value
* - 4: Inhibit suspending the session or computer
* - 12: Inhibit the session being marked as idle
*/
this.inhibitFlags = 12;
```
3. Change `this.inhibitFlags = 12;` to `this.inhibitFlags = 4;`.
Add `/* CHANGED THE ABOVE FROM 12 TO 4 TO INHIBIT SUSPEND BUT ALLOW SCREEN BLANKING */` on the line beneath it to note the change.
4. Find this:
```
if (blankState) {
   this.inhibitFlags = 4;
} else {
   this.inhibitFlags = 4;
}
```
5. Change to this (notes added to reflect change):
```
if (blankState) {
   this.inhibitFlags = 4; // Always allow blank screen
} else {
   this.inhibitFlags = 4; // Do not override to 12; CHANGED FROM 12 TO 4
}
```

6. Logout then log back in, or restart your machine.

Caffeine should now be behaving as expected, allowing the screen to blank but preventing the system from suspending.
